clj/seqs.py

```diff
@@ -655,6 +655,23 @@ def tree_seq(has_branch, get_children, root):
                 yield subchild
 
 
+def dedupe(coll):
+    """
+    Returns a generator of the elements of coll with consecutive duplicates removed.
+    """
+    coll = iter(coll)
+    try:
+        prev = next(coll)
+    except StopIteration:
```

Comment:
You can replace the try/except + a loop with a single loop:
```python3
def dedupe(coll):
    """
    ...
    """
    initial = True
    prev = None

    for e in coll:
        if initial or e != prev:
            initial = False
            yield e

        prev = e
```

Reply:
I didn't want to check that flag each time around the loop. Do you think that isn't a big enough deal?

Reply:
No, I don’t think that’s a big deal.